src/editor/button_widget.cpp

```diff
@@ -107,16 +136,35 @@ ButtonWidget::on_mouse_button_down(const SDL_MouseButtonEvent& button)
 bool
 ButtonWidget::on_mouse_motion(const SDL_MouseMotionEvent& motion)
 {
-  Vector mouse_pos = VideoSystem::current()->get_viewport().to_logical(motion.x, motion.y);
+  m_mouse_pos = VideoSystem::current()->get_viewport().to_logical(motion.x, motion.y);
 
   if (m_grab) {
-    m_hover = m_rect.contains(mouse_pos);
+    m_hover = m_rect.contains(m_mouse_pos);
     return true;
-  } else if (m_rect.contains(mouse_pos)) {
+  } else if (m_rect.contains(m_mouse_pos)) {
     m_hover = true;
     return false;
   } else {
     m_hover = false;
     return false;
   }
 }
+
+void
+ButtonWidget::set_sprite(SpritePtr sprite)
+{
+  m_sprite = std::move(sprite);
+  m_rect.set_size(sprite->get_width() * 1.0f, sprite->get_height() * 1.0f);
```

Comment:
Why? A leftover, perhaps?

```suggestion
  m_rect.set_size(sprite->get_width(), sprite->get_height());
```

Reply:
Does sprite->get_width / get_height() return a float? If not, this converts the value to float.

Reply:
Indeed that is the case. A cast works too but its not a big deal